tests/test_facebook_pagination.py

```diff
@@ -0,0 +1,62 @@
+import test_client as tc
+import time
+import unittest
+
+from tap_tester.base_suite_tests.pagination_test import PaginationTest
+from tap_tester import connections, runner, menagerie, LOGGER
+
+from base_new_frmwrk import FacebookBaseTest
+
+fb_client = tc.TestClient()
+
+
+class FacebookDiscoveryTest(PaginationTest, FacebookBaseTest):
+    """Standard Pagination Test"""
+
+    @staticmethod
+    def name():
+        return "tt_facebook_pagination"
+    def streams_to_test(self):
+        # TODO ads_insights empty for account, no post via API, spike on generating data
+        return {'adcreative', 'ads', 'adsets', 'campaigns'}
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """
+        Setup for tests in this module.
+        """
+        if PaginationTest.synced_records and PaginationTest.record_count_by_stream:
+            return
+
+        # instantiate connection
+        conn_id = connections.ensure_connection(self)
+
+        # run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # table and field selection
+        test_catalogs = [catalog for catalog in found_catalogs
+                         if catalog.get('stream_name') in self.streams_to_test()]
+
+        # non_selected_fields are none
+        self.perform_and_verify_table_and_field_selection(conn_id, test_catalogs)
+
+        # ensure there is enough data to paginate
+        for stream in self.streams_to_test():
+            limit = self.expected_page_size(stream)
+            response = fb_client.get_account_objects(stream)
+
+            number_of_records = len(response['data'])
+            if number_of_records >= limit and response.get('paging', {}).get('next'):
```

Comment:
I would think for pagination to occur the number of records cannot be equal to the limit.  If the limit is 100 and there are 100 then it won't go to a second page.

Reply:
I'm not clear on what `response.get('paging', {}).get('next'):` does?

Reply:
It appears as if the limit sent in the get request controls the number of records returned per page so I don't expect to get greater than 'limit' records back in a single response.  I guess we should change this from ">=" to just "==".  The "response.get('paging', {}).get('next'):" verifies that there is another page of data still left to get.

Reply:
This logic should be in the `get_account_objects`.  When the if statement is true we just continue to the next stream.  Are we assuming because there is a next page, we have another record and are over the limit.